overlord/snapstate/autorefresh.go

```diff
@@ -407,6 +413,38 @@ func canContinueAutoRefresh(st *state.State) (int, bool) {
 	return 0, false
 }
 
+func (m *autoRefresh) restoreMonitoring() error {
+	if m.restoredMonitoring {
+		return nil
+	}
+
+	var monitored []string
+	if err := m.state.Get("monitored-snaps", &monitored); err != nil && !errors.Is(err, state.ErrNoState) {
+		return err
+	}
+
+	defer func() { m.restoredMonitoring = true }()
+	if len(monitored) == 0 {
+		return nil
+	}
+
+	monitoring := make(map[string]chan<- bool)
+	for _, snap := range monitored {
+		done := make(chan string, 1)
+		if err := cgroupMonitorSnapEnded(snap, done); err != nil {
+			return fmt.Errorf("cannot monitor for snap closure: %w", err)
```

Comment:
Is the intention to stop refresh monitoring for all monitored snaps when encountering cgroup monitoring error for any snap?

Reply:
I changed this in the latest commit to just restore what it can and log the rest. Ideally, we would save some in-between state and retry later but that's not possible since this has to be done in an Ensure. In any case, if errors happen here, we probably have bigger issues so this should be fine for now